src/master-playlist-controller.js

```diff
@@ -361,6 +370,32 @@ export class MasterPlaylistController extends videojs.EventTarget {
         } else {
           addSeekableRange();
         }
+
+        buffered = this.tech_.buffered();
+        currentTime = this.tech_.currentTime();
+
+        if (buffered.length) {
+          lastBufferedEnd = buffered.end(buffered.length - 1);
+        }
+        bufferedTime = lastBufferedEnd - currentTime;
+
+        // the end of the current playlist
+        playEnd = this.playlistEnd(updatedPlaylist);
+
+        // the time between end of the playlist and the end of the buffered
+        endTime = playEnd - lastBufferedEnd;
+
+        if (this.seekable().length) {
+          seekableEnd = this.seekable().end(0);
+        }
+
+        // blacklist this playlist if the playhead is at the end of the buffer
+        // and the buffered data ends at the end of the last segment in the playlist
+        if (bufferedTime <= Ranges.TIME_FUDGE_FACTOR &&
+            endTime <= Ranges.TIME_FUDGE_FACTOR &&
+            currentTime > seekableEnd) {
```

Comment:
`seekableEnd`, `lastBufferedEnd` and `endTime` have the potential of being NaN at this point since `seekableEnd` and `lastBufferedEnd` may never be assigned a value if the conditionals for the if statements fail, and `playEnd` may be null from `this.playlistEnd()` affecting the calculation of `endTime`